packages/schemas/alterations/next-1673001922-support-generic-passcode.ts

```diff
@@ -0,0 +1,22 @@
+import { sql } from 'slonik';
+
+import type { AlterationScript } from '../lib/types/alteration.js';
+
+const alteration: AlterationScript = {
+  up: async (pool) => {
+    await pool.query(sql`
+      alter table passcodes alter column interaction_jti drop not null;
+      create index passcodes__email_type on passcodes (email, type);
+      create index passcodes__phone_type on passcodes (phone, type);
+    `);
+  },
+  down: async (pool) => {
+    await pool.query(sql`
+      drop index passcodes__email_type;
+      drop index passcodes__phone_type;
+      alter table passcodes alter column interaction_jti set not null;
```

Comment:
should we drop those null rows or set their `jti` to `''` or something?

Reply:
Ah, very good point. I think we should just simply remove those rows. I also think that we should clear the passcode once it is consumed since it doesn't seem to have any value any more

Reply:
Script updated.